### DISCUSSION TITLE
Multiple Prisma instances with multiple database servers

### DISCUSSION THREAD

**Discussion by cyrilchapon** [prisma/prisma#16749] [Q&A]
Hi,

I suspect this will be controversial but, well 🙂

In a monorepo of my own, I'd like to be able to do the following :
- `packageA` => holds a schema.prisma connected to a databaseA
- `packageB` => holds a schema.prisma connected to a databaseB
- `packageC` => imports both `packageA` and `packageB`

I couldn't find any decent discussion around this. Everything is about Multiple Prisma instance on a single database OR a single Prisma instance handling multiple database / schemas.

My use-case is way simpler; I don't need to mix and match stuff, I don't need shared models accross databases or something like that. What I need is separation of concerns, separated migrations journeys, separated connections pools, separated everything.

(I can expand on the use-case if you ask).

I haven't tried this deeply, but I'm pretty sure the way prisma is handled through npm modules is gonna break something when I'll import `packageA` and `packageB`, if that makes sense.

Any guidelines on this ?

**Comment by nurul3101** [comment]
Hello @cyrilchapon 👋 

Did you have a look at this example of [Monorepo-prisma](https://github.com/nikolasburk/monorepo-prisma)?
It looks like you want to achieve a bit similar by importing PrismaClient Instance in other packages. Is there any specific issue which you are running into?

Also, if it helps, here is an additional example of using [Turborepo with Prisma](https://github.com/vercel/turbo/tree/main/examples/with-prisma).

**Comment by cyrilchapon** [comment]
Hey @nurul3101 ,

I basically know how to wrap prisma in a subpackage; but my goal is to have 2 of them.
To expand on this example my goal would be to have

```
- prisma1
- prisma2
- server (imports prisma1 and prisma2)
```

**Reply by nurul3101** [reply]
I would recommend you to have a look at this [Article](https://dev.to/sabinthedev/nx-prisma-generator-af3). It is written by a Prisma Employee, and the article intends to achieve the same goal which you are trying to achieve.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
